### PR TITLE
Add query decomposition example

### PR DESCRIPTION
**Pull request by lucabeetz**
Made new PR because of weird merge.

This adds an example for the query decomposition, it finally worked with GPT-4 turbo and JSON mode. Had some failures in the old PR #37

### CODE REVIEW

File: examples/query_decomposition/index.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+type QueryPlan = z.infer<typeof QueryPlanSchema>
+
+const oai = new OpenAI({
+  apiKey: process.env.OPENAI_API_KEY ?? undefined,
+  organization: process.env.OPENAI_ORG_ID ?? undefined
+})
+
+const client = Instructor({
+  client: oai,
+  mode: "JSON",
```
Comment (jxnl) [inline examples/query_decomposition/index.ts:27]:
```suggestion
  mode: "FUNCTIONS",
```

**Comment by jxnl** [comment]
can you also add to examples

File: docs/examples/query_decomposition.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
+# Planning and Executing a Query Plan
```
Comment (jxnl) [inline docs/examples/query_decomposition.md:1]: make sure to update mkdocs.yml in the root directory so you can link to it

Comment (lucabeetz) [reply docs/examples/query_decomposition.md:1]: updated

File: docs/examples/index.md [hunk 3]
```diff
@@ -6,4 +6,5 @@
 
 ## Table of Contents
 
-- [How do I do classification?](./classification.md)
\ No newline at end of file
+- [How do I do classification?](./classification.md)
+- [How can I decompose queries?](./query_decomposition.md)
```
Comment (jxnl) [inline docs/examples/index.md:10]:
```suggestion
- [How are complex queries decomposed into subqueries for a single request?](./query_decomposition.md)
```

**Comment by jxnl** [comment]
dope just try 

`mkdocs serve -w .`

take a preview and then rerequest and I'll merge it

**Comment by lucabeetz** [comment]
works!
Screenshot 2024-01-04 at 17 16 28

**Pull request merged**